### PR TITLE
Remove OOB Commands and Pdbs from stage2

### PR DESCRIPTION
**Pull request by brthor**
remove out of box commands and all pdbs from stage2 so they will not be included in our installers.

cc @Petermarcu @Sridhar-MS @piotrpMSFT

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2151)

### CODE REVIEW

File: scripts/run-build.ps1 [hunk 1]
```diff
@@ -72,7 +72,7 @@ $env:PATH = "$env:DOTNET_INSTALL_DIR;$env:PATH"
 # Restore the build scripts
 Write-Host "Restoring Build Script projects..."
 pushd $PSScriptRoot
-dotnet restore --disable-parallel
+dotnet restore --disable-parallel --infer-runtimes
```
Comment (brthor) [inline scripts/run-build.ps1]: cc @pakrym build is breaking without this, did you forget to add it here is there a different issue?

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 2]
```diff
@@ -214,9 +212,31 @@ public static BuildTargetResult CompileStage2(BuildTargetContext c)
                 }
             }
 
+            CleanOutputDir(Path.Combine(Dirs.Stage2, "sdk"));
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Lets clean it from Stage1 too.

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 3]
```diff
@@ -214,9 +212,31 @@ public static BuildTargetResult CompileStage2(BuildTargetContext c)
...
+                {
+                    FS.Rm(file);
+                }
+            }
+
+            // Clean up all .pdb files
+            var pdbs = Directory.EnumerateFiles(directory, "*.pdb", SearchOption.AllDirectories);
+            foreach (var pdb in pdbs)
+            {
+                FS.Rm(pdb);
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Lets create an utility which clean given files from a directory.

Comment (brthor) [inline scripts/dotnet-cli-build/CompileTargets.cs]: :+1:

File: scripts/dotnet-cli-build/CompileTargets.cs (hunk 2 above)
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Actually I am a bit hesitant on cleaning out the PDBs from stage2 or stage1. How do we debug the dotnet CLI tools in our dev box with the PDBs missing?

Comment (brthor) [inline scripts/dotnet-cli-build/CompileTargets.cs]: pubs will still be in the sdk compilation folder right? Could copy them over.

I guess we could change packaging not to pick up stage2 and create a special packaging stage2 as an alternative.

Comment (brthor) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Made the changes for the utility and stage1 removal, looks much nicer now.

**Pull request merged**